source/_integrations/template.markdown

```diff
@@ -339,6 +339,20 @@ If you are using the state of a platform that might not be available during star
 
 {% endraw %}
 
+### Template errors
+
+If any of the templates generate an error when rendered, the sensor will become `unavailable`.
+
+### Caveats using the `this` variable
+
+The `this.state` data will be initialized with `unknown` during startup. 
+
+The `this.attributes` will be initialized with the empty mapping `{}` during startup. For that reason, using any attribute in your template, such as `this.attributes.my_attribute` will result in an `UndefinedError` during startup. To avoid this, use `state_attr(this.entity_id, 'my_attribute')` instead.
```

Comment:
This is odd advise, as the provided solution is different compared to what the original code did.

Reply:
You mean `this.attributes.my_attribute` is different from `state_attr(this.entity_id, 'my_attribute')` in that the former is fixed to the trigger-time, while the latter _might_ change? While that is true, of course, it will likely not make a difference in template entities, because they don't have any delays. 

The obvious alternative is to use `if`, but it will become clumsy.

Would you be fine to point out both alternatives and describe their difference?

(The other difference between the suggestion and the original code is that the suggestion guards against undefined attributes, but that is intended.)

Reply:
> You mean this.attributes.my_attribute is different from state_attr(this.entity_id, 'my_attribute') in that the former is fixed to the trigger-time, while the latter might change? While that is true, of course, it will likely not make a difference in template entities, because they don't have any delays.

That is not entirely true. Anyways, they are not the same thing :)

Reply:
I agree. But we'll have to change something, if we want to avoid the error. ;-)

What would you suggest?

Reply:
> But we'll have to change something, if we want to avoid the error.

What error? If you handle it right, there is no error? I'm lost on why that difference is needed. When accessing an attribute from a dictionary, one needs to ensure the key exists, otherwise, an KeyError will be thrown.

The described solution, is not a solution, but a workaround. The solution, is to ensure a key exists before accessing it on a dict.

Reply:
During startup (i.e. before the first evaluation of the template sensor) there will be no attributes in `this`. So, there's no way for the user to make sure the key exists. And that results in `UndefinedError`s, when accessing the attributes.

(Unless we change the `core` code to initialize all defined attributes with `None`. Then, we wouldn't need the workaround anymore. Which might be a good idea in the medium run.)

It is really only an issue during startup. See the second halve of this comment, e.g.: https://github.com/home-assistant/core/issues/78358#issuecomment-1267127022 . Here, it doesn't harm, but it bothers. Maybe there are more involved examples, which will break the sensor permanently. (I tried to come up with an example, but I cannot test it currently, so, I'll rather not guess into the wild.)